reconcile/database_access_manager.py

```diff
@@ -0,0 +1,506 @@
+                                {
+                                    "name": "configs",
+                                    "mountPath": "/tmp/scripts/",
+                                    "readOnly": True,
+                                },
+                            ],
+                        },
+                    ],
+                    "volumes": [
+                        {
+                            "name": "configs",
+                            "projected": {
+                                "sources": [
+                                    {
+                                        "secret": {
+                                            "name": job_data.script_secret_name,
+                                        },
+                                    },
+                                ],
+                            },
+                        },
+                    ],
+                },
+            },
+        },
+    }
+    return OpenshiftResource(
+        body=job,
+        integration=QONTRACT_INTEGRATION,
+        integration_version=QONTRACT_INTEGRATION_VERSION,
+    )
+
+
+def get_service_account_spec(name: str) -> OpenshiftResource:
+    return OpenshiftResource(
+        body={
+            "apiVersion": "v1",
+            "kind": "ServiceAccount",
+            "metadata": {
+                "name": name,
+                "labels": {
+                    "app": "qontract-reconcile",
+                    "integration": QONTRACT_INTEGRATION,
+                },
+            },
+            "automountServiceAccountToken": False,
+        },
+        integration=QONTRACT_INTEGRATION,
+        integration_version=QONTRACT_INTEGRATION_VERSION,
+    )
+
+
+class DBAMResource(BaseModel):
+    resource: OpenshiftResource
+    clean_up: bool
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class JobStatusCondition(BaseModel):
+    type: str
+
+
+class JobStatus(BaseModel):
+    conditions: list[JobStatusCondition]
+
+    def is_complete(self) -> bool:
+        return True if self.conditions else False
+
+    def has_errors(self) -> bool:
+        for condition in self.conditions:
+            if condition.type == "Failed":
+                return True
+        return False
+
+
+class DatabaseAccessManagerIntegration(QontractReconcileIntegration):
+    @property
+    def name(self) -> str:
+        return QONTRACT_INTEGRATION
+
+    def _populate_resources(
+        self,
+        db_access: DatabaseAccessV1,
+        engine: str,
+        image_repository: str,
+        pull_secret: dict[any, any],
+        resource: NamespaceTerraformResourceRDSV1,
+        resource_name: str,
+        settings: dict[any, any],
+    ) -> list[DBAMResource]:
+        managed_resources: list[DBAMResource] = []
+
+        # create service account
+        managed_resources.append(
+            DBAMResource(
+                resource=get_service_account_spec(resource_name),
```

Comment:
do you want to prefix the SA name with something to prevent potential conflicts with other SAs that exist in the namespace already?

Reply:
The resource_name is "{dbam-dbam_resource.name}". That should make it unique